### ISSUE TITLE
Move `conda` out of `base` env

### ISSUE THREAD

**Issue by kenodegard** [conda/conda#11342]
### Summary

There is a long history of issues where users install anything and everything into their `base` environment (previously called the `root` environment) which invariably at some point (or after running `conda update conda`) results in some `conda` dependency being upgraded (or downgraded) and `conda` itself breaking. Efforts have been made in the past to circumvent this issue by vendoring `conda`'s dependencies (see https://github.com/conda/conda/issues/2825) but to this day `conda` is still not fully siloed and independent.

The idea for splitting `conda` out of the `base`/`root` environment is an old one (predating 2016) and has been earmarked as a conda 5.0 feature for quite some time.

#### Proposal
1. Define a new "class" of environments: hidden environments
    a. Most likely would just be an environment prefixed with a `.`
    b. Modifying `conda info --envs` and `conda env list` to hide hidden envs by default
4. Allow `conda` to be run/shell functions initialized out of any environment
5. Define a new `conda` command/args for upgrading `conda`:
    ```bash
    conda update --conda

    # the above proposal is effectively the same as the following:
    conda update --prefix HIDDEN_CONDA_ENV conda

    # adding this new `--conda` argument allows us to make a
    # more clean interface for say installing plugins:
    conda install --conda conda-doctor

    # or even to make it convenient to migrate existing installs to the
    # new siloed env setup:
    conda create --conda
    ```
6. Modify installer to install with a base env and a hidden conda env

#### Related ideas
- https://github.com/conda/conda/issues/4929
- https://github.com/conda/conda/issues/4943

**Comment by kenodegard** [comment]
micromamba has already moved in this direction.

> The lack of a "base" environment makes things less brittle vs. traditional conda/mamba."
> https://link.medium.com/dCB8wwNzQob

**Comment by LtDan33** [comment]
This is a great CEP candidate. That would be the next step.

**Comment by jaimergp** [comment]
I've started some experiments in this direction:

- Idea for installers: supply a `default` env (see https://github.com/conda-forge/miniforge/issues/636).
- Idea for `conda`: lock `base` by default ([see this logic](https://github.com/conda/conda/pull/14113/commits/fbfad1610c8bcc14b6f4c36dae3bde45a8f9c8b8)) or, even better, come up with a CEP for PEP-668 style locking (EXTERNALLY-MANAGED for conda).
- A new type of conda-standalone packaging with no extraction overheads. This essentially becomes `base` without being a `conda` environment at all. More details at https://github.com/conda/conda-standalone/pull/93.

**Comment by danpetry** [comment]
I understand that this solves the issue at hand, however I guess that ideally users should be using different environments for each project or project class, for performance and package recency reasons? Have there been any thoughts on a way of very obviously informing people that this would be the best way to use their environments (eg obvious message when they create new environments or something like that?)

**Comment by notatallshaw** [comment]
> I guess that ideally users should be using different environments for each project

Yes, but conda installers (Anaconda and miniforge at least) default the user into to the base environment. So the natural thing for a new user is to start installing there, and it's the one environment that if it breaks can break the entire installation.

There can be a better user experience, i.e. not defaulting users into an environment that can immediately break conda, as well as better education.

**Comment by jaimergp** [comment]
There is an open PR to change which environment gets activated by default, and this could be shipped in the installer config once merged and released, IIRC.

**Comment by jezdez** [comment]
> There is an open PR to change which environment gets activated by default, and this could be shipped in the installer config once merged and released, IIRC.

Which PR is this for record-keeping?

**Comment by danpetry** [comment]
Yes just to be clear on what I'm talking about

> There can be a better user experience, i.e. not defaulting users into an environment that can immediately break conda

This PR and the ones linked here

> as well as better education.

What I'm suggesting. Anyway, I opened an issue [here](https://github.com/conda/conda/issues/14389) as it's not really the topic of this issue.

**Comment by jaimergp** [comment]
One step towards "protecting" the base environment would be to plant a `frozen` marker file that tools would respect and only override explicitly if needed / risks are understood. To that end, I submitted this CEP draft: https://github.com/jaimergp/ceps/blob/frozen-env/cep-00%3F%3F.md

**Comment by notatallshaw** [comment]
> One step towards "protecting" the base environment would be to plant a `frozen` marker file that tools would respect and only override explicitly if needed / risks are understood. To that end, I submitted this CEP draft: [conda/ceps#99](https://github.com/conda/ceps/pull/99)

For Python packaging there already exists a standard which is respected by tools, and it has been long proposed conda adopt it for at least the base environment: https://github.com/conda/conda/issues/12245

It would have saved tools like uv making guesses on what the base environment is: https://github.com/astral-sh/uv/pull/7691

**Comment by jaimergp** [comment]
> For Python packaging there already exists a standard which is respected by tools, and it has been long proposed conda adopt it for at least the base environment: https://github.com/conda/conda/issues/12245

You might like https://github.com/conda-incubator/conda-pypi then :)

**Comment by notatallshaw** [comment]
> You might like https://github.com/conda-incubator/conda-pypi then :)

This doesn't seem to be widely adopted enough to help Python packaging tools (pip, uv, poetry, hetch, etc.). I'm not in personal need of this, and I'm not sure it's mature enough to recommend to newbies who break their conda environment.
